common/lib/container-definitions/src/loader.ts

```diff
@@ -121,8 +121,24 @@ export interface IContainer extends IEventProvider<IContainerEvents>, IFluidRout
 
     /**
      * The current code details for the container's runtime
+     * @deprecated use getSpecifiedCodeDetails for the code details currently specified for this container, or
+     * getUsedCodeDetails for the code details that the container's context was loaded with.
+     * To be removed after getSpecifiedCodeDetails and getUsedCodeDetails become ubiquitous.
      */
-    readonly codeDetails: IFluidCodeDetails | undefined
+    readonly codeDetails: IFluidCodeDetails | undefined;
+
+    /**
+     * Get the code details that are currently specified for the container.
+     * @returns The current code details if any are specified, undefined if none are specified.
+     */
+    getSpecifiedCodeDetails?(): IFluidCodeDetails | undefined;
+
+    /**
+     * Get the code details that were used to load the container.
+     * @returns The code details that were used to load the container if it is loaded, undefined if it is not yet
+     * loaded.
+     */
+    getUsedCodeDetails?(): IFluidCodeDetails | undefined;
```

Comment:
maybe a better name is getRuntimeCodeDetails, as this is the codedetails used by the code loader to get the IRuntimeFactory

Reply:
Another option I was thinking about was `getLoadedCodeDetails`.  I think we're going to do a design review for this one so maybe we can discuss naming options there (I'll hold the PR until then).

For "used" here I was drawing on CSS terminology of [specified](https://developer.mozilla.org/en-US/docs/Web/CSS/specified_value) vs. [used](https://developer.mozilla.org/en-US/docs/Web/CSS/used_value).  Which is not really relevant to mimic, but has a similar distinction of "what is specified might not match what is in use".

Reply:
makes sense. i like loaded, as it comes from the code loader. trying to unify concepts i think will make explanation and docs easier. I think both, "getLoadedCodeDetails | getRuntimeCodeDetails returns the codedetails of the runtime factory loaded by the code loader" are both reasonable statements the link this property to where it comes from